### PR TITLE
Add safestringlib

### PR DESCRIPTION
**Pull request by tjohnston-fcg**
update to use safestringlib for some C standard library functions.

### CODE REVIEW

File: hipflowapp/Hip_Native/timer.cpp [hunk 1]
```diff
@@ -254,10 +259,10 @@ void timinglogmsg(const char *text, uint8_t rspCount, BYTE *rspMsg)
 
 	for (i = 0; i < rspCount; i++)
 	{
-		sprintf(buf+3*i, "%02x ", rspMsg[i]);
+		sprintf_s(buf+3*i, 500, "%02x ", rspMsg[i]);
```
Comment (adowning701) [inline hipflowapp/Hip_Native/timer.cpp]: 500 is a hard coded number. This should should be a variable.

File: hipflowapp/Hip_Native/timer.cpp [hunk 2]
```diff
@@ -254,10 +259,10 @@ void timinglogmsg(const char *text, uint8_t rspCount, BYTE *rspMsg)
...
 	}
 
-	sprintf(buf2, "%s: %s", text, buf);
+	sprintf_s(buf2, 500, "%s: %s", text, buf);
```
Comment (adowning701) [inline hipflowapp/Hip_Native/timer.cpp]: 500 is a hard coded number. This should should be a variable.

File: hipflowapp/Hip_Native/Data/insert_Extract.cpp [hunk 3]
```diff
@@ -237,7 +238,7 @@ uint8_t c_insert_Extract::extract( float   & x, uint8_t **ppData, uint8_t& dataC
 	//uint16_t g = (uint16_t)((x >> 16) & 0xffff);// hi word
```
Comment (adowning701) [inline hipflowapp/Hip_Native/Data/insert_Extract.cpp]: Needs an explanation.

File: hipflowapp/Hip_Native/Data/varient.h [hunk 4]
```diff
@@ -245,14 +245,14 @@ inline ostream &operator<<(ostream& ostr, CValueVarient& vv )
 			
 				if (vv.vIsDouble)
 				{
-			//	sprintf(tmpbuff,"const d %.15f (0x%a)",z,z);
```
Comment (adowning701) [inline hipflowapp/Hip_Native/Data/varient.h:248]: Needs an explanation.

File: hipflowapp/Hip_Native/Files.h [hunk 5]
```diff
@@ -38,7 +38,7 @@ using namespace std::__cxx11;
 extern void nop();
 
 #ifdef _DEBUG
-	#define bcmLOGIT( C... )   printf( C )
+	//#define bcmLOGIT( C... )   printf( C )
```
Comment (adowning701) [inline hipflowapp/Hip_Native/Files.h]: Needs an explanation.

**Review by adowning701 — CHANGES_REQUESTED** [review]
Please address comments.

File: hipflowapp/Hip_Native/timer.cpp (hunk 1 above)
Comment (tjohnston-fcg) [reply hipflowapp/Hip_Native/timer.cpp]: replace 500 with a const int

File: hipflowapp/Hip_Native/timer.cpp (hunk 2 above)
Comment (tjohnston-fcg) [reply hipflowapp/Hip_Native/timer.cpp]: replace with const int

File: hipflowapp/Hip_Native/Data/insert_Extract.cpp (hunk 3 above)
Comment (tjohnston-fcg) [reply hipflowapp/Hip_Native/Data/insert_Extract.cpp]: remove the code except the last line, which needed to be retained.  i tested it.

File: hipflowapp/Hip_Native/Data/varient.h (hunk 4 above)
Comment (tjohnston-fcg) [reply hipflowapp/Hip_Native/Data/varient.h:248]: all this code is wrapped in an ifdef.

File: hipflowapp/Hip_Native/Files.h (hunk 5 above)
Comment (tjohnston-fcg) [reply hipflowapp/Hip_Native/Files.h]: remove ifdef

**Review by adowning701 — APPROVED** [review]
All requested comments have been addressed or explained.

**Pull request merged**